optuna/storages/_redis.py

```diff
@@ -104,7 +105,7 @@ def __init__(
         self._redis = redis.Redis.from_url(url)
         self._heartbeat_interval = heartbeat_interval
         self._grace_period = grace_period
-        self._failed_trial_callback = failed_trial_callback
+        self.failed_trial_callback = failed_trial_callback
```

Comment:
Why do you change the name of this attribute?

Reply:
Because previously we used ```_failed_trial_callback``` only for ```RedisStorage```, but now it is also used by ```_CachedStorage```.

Reply:
Now I modifyed ```get_failed_trial_callback``` of ```_CachedStorage``` to call ```get_failed_trial_callback``` of ```_backend``` instead. Because it is more reasonable and increases test coverage.

Reply:
Sounds reasonable.